### PR TITLE
add UNETR 3d segmentation tutorial

### LINKED ISSUE #262: Add Tutorial for UNETR (3D Multi-organ Segmentation)
**Is your feature request related to a problem? Please describe.**
Add tutorial for UNETR using 3D multi-organ Segmentation task with BTCV challenge dataset.  

**Describe the solution you'd like**
A notebook which contains all steps for training and validation addresses this feature request.

### PR DESCRIPTION
**Pull request by ahatamiz**
Signed-off-by: ahatamizadeh <ahatamizadeh@nvidia.com>

Fixes #262 

### Description
This pull request adds the 3D multi-organ segmentation tutorial using BTCV challange dataset based on UNETR:
- UNETR based on: "Hatamizadeh et al., UNETR: Transformers for 3D Medical Image Segmentation (https://arxiv.org/abs/2103.10504)"

The BTCV challenge dataset:
https://www.synapse.org/#!Synapse:syn3193805/wiki/217752

### Status
**Ready**

### Checks

- [x] Notebook runs automatically `./runner [-p <regex_pattern>]`

### CODE REVIEW

**Comment by ahatamiz** [comment]
@wyli @Nic-Ma

**Comment by wyli** [comment]
thanks @ahatamiz, could you help append this new filename to the list https://github.com/Project-MONAI/tutorials/blob/d340613cea8b0f10b780694c636c1d7ff8f7658d/runner.sh#L22-L33 Effectively it'll be excluded from auto testing.

File: 3d_segmentation/unetr_segmentation_3d.ipynb [hunk 1]
```diff
@@ -0,0 +1,4317 @@
+{
```
Comment (Nic-Ma) [inline 3d_segmentation/unetr_segmentation_3d.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    img_name = val_ds[case_num]['image_meta_dict']['filename_or_obj'].split('/')[-1]
</pre><p>Suggest to use <code>os.path.split()</code>  to support non-linux path separators.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/263/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (Nic-Ma) [inline 3d_segmentation/unetr_segmentation_3d.ipynb:1]: <p>Could you please help review the output of this cell? It's some info of your own hardware.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/263/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (Nic-Ma) [inline 3d_segmentation/unetr_segmentation_3d.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #68.    model.load_state_dict(torch.load(os.path.join(root_dir, "best_metric_model.pth")))
</pre><p>Could you please help delete the output of this cell? It's too line in the notebook and you already plot the loss curve and validation metrics curve, I think we don't need to keep the detailed log.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/263/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (Nic-Ma) [inline 3d_segmentation/unetr_segmentation_3d.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.            if img_name=='img0039.nii.gz':
</pre><p>If you only print 1 image, I think we don't need a for loop here?</p><p> </p><p>Thanks.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/263/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

**Comment by ahatamiz** [comment]
> thanks @ahatamiz, could you help append this new filename to the list
> 
> https://github.com/Project-MONAI/tutorials/blob/d340613cea8b0f10b780694c636c1d7ff8f7658d/runner.sh#L22-L33
> 
> Effectively it'll be excluded from auto testing.

Thanks @wyli. Added.

File: 3d_segmentation/unetr_segmentation_3d.ipynb (hunk 1 above)
Comment (ahatamiz) [reply 3d_segmentation/unetr_segmentation_3d.ipynb]: Thanks @Nic-Ma. Fixed.

Comment (ahatamiz) [reply 3d_segmentation/unetr_segmentation_3d.ipynb]: Fixed.

Comment (ahatamiz) [reply 3d_segmentation/unetr_segmentation_3d.ipynb]: Fixed.

Comment (ahatamiz) [reply 3d_segmentation/unetr_segmentation_3d.ipynb]: Fixed.

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update.
Looks good to me now.

Thanks.

**Comment by wyli** [comment]
in case you didn't see it, there are some pep8 errors in the notebook: https://github.com/Project-MONAI/tutorials/pull/263/checks?check_run_id=3067846658#step:7:83

**Comment by ahatamiz** [comment]
> in case you didn't see it, there are some pep8 errors in the notebook: https://github.com/Project-MONAI/tutorials/pull/263/checks?check_run_id=3067846658#step:7:83

T

> Thanks for the quick update.
> Looks good to me now.
> 
> Thanks.

Thanks @Nic-Ma

**Comment by ahatamiz** [comment]
> in case you didn't see it, there are some pep8 errors in the notebook: https://github.com/Project-MONAI/tutorials/pull/263/checks?check_run_id=3067846658#step:7:83

Thanks @wyli . I fixed it.

**Comment by wyli** [comment]
there are still some errors. because it's your first PR to this repo, the pipeline wouldn't run automatically... I can help trigger it  when you update the PR

**Comment by ahatamiz** [comment]
> there are still some errors. because it's your first PR to this repo, the pipeline wouldn't run automatically... I can help trigger it when you update the PR

I see. Sorry about that. Currently working on fixing the errors. Is there any local tool I can use to perform these checks before updating the PR?

**Comment by wyli** [comment]
the pipeline runs these commands from the tutorial repo with a ubuntu linux
```
python -m pip install --upgrade pip wheel
python -m pip install -r requirements.txt
./runner.sh --no-run
```
but I couldn't run it locally with a Mac, perhaps you could try it as well

**Comment by ahatamiz** [comment]
> the pipeline runs these commands from the tutorial repo with a ubuntu linux
> 
> ```
> python -m pip install --upgrade pip wheel
> python -m pip install -r requirements.txt
> ./runner.sh --no-run
> ```
> 
> but I couldn't run it locally with a Mac, perhaps you could try it as well

Thanks @wyli.

**Comment by ahatamiz** [comment]
Hi @wyli, on top of addressing the issues, I made additional formatting using `black`. Hopefully these will address all errors.
